todo-list-connected-to-api/maryam/src/App.js

```diff
@@ -0,0 +1,80 @@
+import React, {useState, useEffect} from 'react';
+import {
+  makeStyles,
+  Grid,
+  List,
+  Typography,
+} from '@material-ui/core';
+// ----------------------------------------------------------------
+import AddNewItem from './AddNewItem';
+import TodoItem from './TodoItem';
+// ----------------------------------------------------------------
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width:'100%',
+    height: '100vh',
+    overflowX: 'hidden',
+    backgroundPositionX: '100%',
+    backgroundPositionY: '100%',
+    backgroundColor: '#3f3160',
+  },
+  container:{
+    width:'40%',
+    height: '70%',
+    overflowX: 'hidden',
+    backgroundColor: theme.palette.common.white,
+    boxShadow:theme.shadows[15],
+  },
+  todolist:{
+    width: '100%',
+    backgroundColor: theme.palette.background.paper,
+  }
+}));
+// ----------------------------------------------------------------
+const App = () => {
+  const classes = useStyles();
+  const [todoItems, setTodoItems] = useState([]);
+
+  useEffect(() => {
+    fetch(`https://todolist.ehsan-rafee.ir/api/todolist`,{
+      method: 'GET',
+      headers: {
+          'Content-Type':'application/json'
+      }
+    })
+    .then(response => response.json())
+    .then(data => setTodoItems(data.flatMap(e => e.ownerId === 5 ? e : [])))
+    .catch(error => console.error(error))}
+  ,[todoItems]);
```

Comment:
__[HIGH]__  
Think about it: What are you syncing this `fetch` call with? That is, which one of the following you think is true?
(With _initializing items_ I mean calling `fetch`. With _sync A with B_ I mean do A whenever B happens.)  

1. Sync initializing items with component mount
2. Sync initializing items with to-do items updates

The answer is 1 for sure. Choosing 2 is irrational.

Reply:
I think I now understand what you mean 🤔 
By handling the deletion and editing of items in the FETCH related to themselves, this issue becomes OK... Am I right? 🤔

Reply:
Almost right. When you handle setting states manually (in `fetch(...).then` or anywhere else), you don't need to add something like `[todoItems]` dependency to your `useEffect`.